File: scripts/vcf2gene_info.pl
#!/usr/bin/perl -w
use strict;

my $gene_lookup = $ARGV[0];
my $infile = $ARGV[1];
my $header_flag = 1;

my %GeneLookup;
my @GeneOrder;
my %Sizes;
my %Results;

open(IN,"<$gene_lookup") || die "\n Cannot open the file: $gene_lookup\n";
while(<IN>) {
    chomp;
    unless ($_ =~ m/^#/) {
	my @a = split(/\t/, $_);
	push(@GeneOrder, $a[0]);
	if (exists $GeneLookup{$a[1]}{'1'}) {
	    for (my $i=$a[3]; $i<= $a[4]; $i++) {
		$GeneLookup{$a[1]}{$i} = $GeneLookup{$a[1]}{$i} . "," . $a[0];
		$Sizes{$a[0]}++;
	    }
	}
	else {
	    for (my $i=1; $i<= $a[2]; $i++) {
		$GeneLookup{$a[1]}{$i} = "non-coding";
		$Sizes{'non-coding'}++;
	    }
	    for (my $i=$a[3]; $i<= $a[4]; $i++) {
		$GeneLookup{$a[1]}{$i} = $a[0];
		$Sizes{'non-coding'}--;
		$Sizes{$a[0]}++;
	    }
	}
    }
}
close(IN);
push(@GeneOrder, "non-coding");

open(IN,"<$infile") || die "\n Cannot open the file: $infile\n";
while(<IN>) {
    chomp;
    if ($_ =~ m/^#CHROM/) {
	$header_flag = 0; ## This is the last line of the header information
	print $_ . "\n";
    }
    elsif ($header_flag == 0) {
	my @a = split(/\t/, $_);
	$a[2] = $GeneLookup{$a[0]}{$a[1]};
	print join("\t", @a) . "\n";
    }
}
close(IN);

exit 0;

File: scripts/Version_0/vcf2gene_info.py
#!/usr/bin/python                                                                                                                                    
import sys

gene_table = open(sys.argv[1], "r")
file_vcf = open(sys.argv[2], "r")

header = True
geneLookup = {}

for line in gene_table:
    geneLine=[]
    if line.rstrip()[0]!="#":
        geneLine = line.rstrip().split('\t')
        for i in [2, 3, 4]:
            geneLine[i]=int(geneLine[i])
        if geneLine[1] in geneLookup:
            for i in range(geneLine[3], geneLine[4]+1):
                geneLookup[geneLine[1]][i]=geneLookup[geneLine[1]][i]+","+geneLine[0]

        else:
            geneLookup[geneLine[1]]={}
            for i in range(1, geneLine[2]+1):
                geneLookup[geneLine[1]][i]='non-coding'
            for i in range(geneLine[3], geneLine[4]+1):
                geneLookup[geneLine[1]][i]=geneLine[0]

gene_table.close()

for line in file_vcf:
    geneLine=[]
    geneLine = line.rstrip().split('\t')
    if geneLine[0]=='#CHROM':
        header = False
        print '\t'.join(geneLine)
    else:
        if header == False:
            geneLine[1]=int(geneLine[1])
            geneLine[2]=geneLookup[geneLine[0]][geneLine[1]]
            print '\t'.join([str(x) for x in geneLine])

file_vcf.close()
